nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -13,6 +15,134 @@
 PathType = Union[str, Path]
 
 
+class VideoCaptureContext:
+    def __init__(self, *args, stub=False, **kwargs):
+        self.vc = cv2.VideoCapture(*args, **kwargs)
+        self._args = args
+        self._kwargs = kwargs
+        self.stub = stub
+        self._current_frame = 0
+        self.frame_count = self.get_movie_frame_count()
+        assert self.frame_count > 0, "movie contains no frames"
+        self.fps = self.get_movie_fps()
+        self.frame = self.get_movie_frame(0)
+        assert self.frame is not None, "unable to read the movie file provided"
```

Comment:
```suggestion
    def __init__(self, file_path: FilePathType, **kwargs):
        self.file_path = str(file_path)

        self.fps = self.get_movie_fps()
        self.frame_count = self.get_movie_frame_count()
        assert self.frame_count > 0, "The movie contains no frames!"
        self._current_frame = 0

        self.frame = self.get_movie_frame(self._current_frame)
        assert self.frame is not None, "Unable to read the movie file provided."

        self.vc = cv2.VideoCapture(filename=self.file_path)
```

It is probably in our best interest to avoid future issues to assume full control over the input arguments to the `VideoCapture`.

Also, in the current version of `nwb-conversion-tools`, we try to always refer to file based input arguments as `file_path`, and these lines would be the ideal place to perform the aliasing.

Also, I think the `stub` option would be more appropriate for the `run_conversion` step of the `MovieDataInterface`, so we can wait to perform it there and just have the context be in charge of general I/O to movie files.

Reply:
yes, it makes sense to control the input arguments to the context with only `file_path` being allowed. I have made changes https://github.com/catalystneuro/nwb-conversion-tools/pull/392/commits/4536888fedb0caed486a19ddb8b392491da03d95